src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TagBuilder.cs

```diff
@@ -117,7 +119,7 @@ private void AppendAttributes(StringBuilder sb)
                     continue;
                 }
 
-                var value = WebUtility.HtmlEncode(attribute.Value);
+                var value = _htmlEncoder.HtmlEncode(attribute.Value);
```

Comment:
Didn't address this comment yet. Had some weird build issues. So updated the PR with rest of the fixes.